racket/src/ChezScheme/s/cptypes-lattice.ss

```diff
@@ -54,6 +55,35 @@
     (nongenerative #{pred-$record/ref zc0e8e4cs8scbwhdj7qpad6k3-1})
     (sealed #t))
 
+  (include "base-lang.ss")
+  (with-output-language (Lsrc Expr)
+    (define void-rec `(quote ,(void)))
+    (define true-rec `(quote #t))
+    (define false-rec `(quote #f))
+    (define null-rec `(quote ()))
+    (define eof-rec `(quote #!eof))
+    (define bwp-rec `(quote #!bwp)))
+
+  (define true-pred (make-pred-or '$immediate/true/nofixnum 'normalptr '$record))
+  (define ptr-pred (make-pred-or '$immediate/nofixnum 'normalptr '$record))
+  (define null-or-pair-pred (make-pred-or null-rec 'pair 'bottom))
+  (define $immediate-pred (make-pred-or '$immediate/nofixnum 'fixnum 'bottom))
+  (define maybe-number-pred (make-pred-or false-rec 'number 'bottom))
+  (define maybe-fixnum-pred (make-pred-or false-rec 'fixnum 'bottom))
+
+  ; This can be implemented with implies?
+  ; but let's use the straightforward test.
+  (define (predicate-is-ptr? x)
+    (and (pred-or? x)
+         (eq? (pred-or-imm x) '$immediate/nofixnum)
+         (eq? (pred-or-imm x) 'normalpred)
+         (eq? (pred-or-imm x) '$record)))
```

Comment:
should be `pred-or-nor` and `pred-or-rec` respectively?

Reply:
Thanks. Fixed.

Reply:
Fixed also `'normalpred` -> `'normalptr`.